hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java

```diff
@@ -243,11 +243,13 @@ public void initWanReplication(NodeEngine nodeEngine) {
      */
     private boolean hasPublisherWithMerkleTreeSync(Config config, String wanReplicationRefName) {
         WanReplicationConfig replicationConfig = config.getWanReplicationConfig(wanReplicationRefName);
-        for (WanPublisherConfig publisherConfig : replicationConfig.getWanPublisherConfigs()) {
-            if (publisherConfig.getWanSyncConfig() != null
-                    && ConsistencyCheckStrategy.MERKLE_TREES.equals(publisherConfig.getWanSyncConfig()
-                                                                                   .getConsistencyCheckStrategy())) {
-                return true;
+        if (replicationConfig != null) {
+            for (WanPublisherConfig publisherConfig : replicationConfig.getWanPublisherConfigs()) {
+                if (publisherConfig.getWanSyncConfig() != null
+                        && ConsistencyCheckStrategy.MERKLE_TREES.equals(publisherConfig.getWanSyncConfig()
```

Comment:
seems `publisherConfig.getWanSyncConfig()` can also be null?

Reply:
`publisherConfig.getWanSyncConfig()` may return `null`, but it is null-checked before referenced. 
`replicationConfig.getWanPublisherConfigs()` can't return null based on the code, null-check can be added though.
Maybe I miss your point?

Reply:
no, you are right, i missed that check.